document/cloud-client/analyze_form.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def analyze_form(project_id, input_uri):
+    """Parse a form"""
+    # [START analyze_form]
+    from google.cloud import documentai_v1beta2 as documentai
+    from google.cloud.documentai_v1beta2 import types
+
+    client = documentai.DocumentUnderstandingServiceClient()
+
+    gcs_source = types.GcsSource(uri=input_uri)
+    input_config = types.InputConfig(
+        gcs_source=gcs_source, mime_type='application/pdf')
+
+    # Improve form parsing results by providing key-value pair hints.
+    # For each key hint, key is text that is likely to appear in the
+    # document as a form field name (i.e. "DOB").
+    # Value types are optional, but can be one or more of:
+    # ADDRESS, LOCATION, ORGANIZATION, PERSON, PHONE_NUMBER, ID,
+    # NUMBER, EMAIL, PRICE, TERMS, DATE, NAME
+    key_value_pair_hints = [
+        types.KeyValuePairHint(key='Name', value_types=['NAME']),
+        types.KeyValuePairHint(key='Agreement Number', value_types=['NUMBER'])
+    ]
```

Comment:
Does this have a noticeable improvement or is it worth trying to split this up into multiple snippets to say, hey here is basic form parsing and here is some advanced tactics to get better results?

Reply:
Your average user is going to have to use these hints if they want decent quality, so I'd say it's worth including here rather than have someone get frustrated with bad quality and not bother to find the "advanced" sample.